src/MorseL.Client.WebSockets/WebSocketClient.cs

```diff
@@ -55,6 +55,8 @@ public WebSocketClient(string uri, Action<WebSocketClientOption> config=null, Ac
             };
             _internalWebSocket.Closed += (sender, args) =>
             {
+                // We're closing, kill the receive loop (which causes listeners to break out)
+                _internalCts.Cancel();
```

Comment:
This coupled with leaving some listeners fixes the issue. The rest is more unit test cruft.